test/dynamo/test_repros.py

```diff
@@ -2335,6 +2335,19 @@ def f(x):
         )
         self.assertEqual(gm(inp).shape, f(inp).shape)
 
+    @torch._dynamo.config.patch("dynamic_shapes", True)
+    def test_dynamic_shapes_implicit_guard(self):
+        def f(x):
+            y = x * x.size(x.shape[0])
+            torch.sum(y, [y.shape[0]])
+            return y
+
+        cnt = torch._dynamo.testing.CompileCounter()
+        opt_fn = torch._dynamo.optimize(cnt, nopython=True)(f)
```

Comment:
is your test only caring about nopython not raising?

i thought i'd see a second call to opt_fn and a recompile to prove the guard worked

Reply:
I mostly only wanted to show that the test got to the end without erroring or graph breaking. I wasn't intending to exercise the guard machinery, although I suppose I could add a second call if you really want

Reply:
no need, just wasn't sure what you were testing